src/output_writers.jl

```diff
@@ -270,126 +270,215 @@ end
-        "yF" => collect(model.grid.yF),
-        "zF" => collect(model.grid.zF),
-        "u" => Array(parentdata(model.velocities.u)),
-        "v" => Array(parentdata(model.velocities.v)),
-        "w" => Array(parentdata(model.velocities.w)),
-        "T" => Array(parentdata(model.tracers.T)),
-        "S" => Array(parentdata(model.tracers.S))
+        "xF" => collect(model.grid.xF)[1:end-1],
+        "yF" => collect(model.grid.yF)[1:end-1],
+        "zF" => collect(model.grid.zF)[1:end-1]
+    )
+    dim_attrib = Dict(
+        "xC" => ["longname" => "Locations of the cell centers in the x-direction.", "units" => "m"],
+        "yC" => ["longname" => "Locations of the cell centers in the y-direction.", "units" => "m"],
+        "zC" => ["longname" => "Locations of the cell centers in the z-direction.", "units" => "m"],
+        "xF" => ["longname" => "Locations of the cell faces in the x-direction.",   "units" => "m"],
+        "yF" => ["longname" => "Locations of the cell faces in the y-direction.",   "units" => "m"],
+        "zF" => ["longname" => "Locations of the cell faces in the z-direction.",   "units" => "m"]
     )
 
-    if fw.async
-        # Execute asynchronously on worker 2.
-        i = model.clock.iteration
-        @async remotecall(write_output_netcdf, 2, fw, fields, i)
-    else
-        write_output_netcdf(fw, fields, model.clock.iteration)
+    # Applies slices to the dimensions d
+    for (d, slice) in slice_kw
+        if String(d) in keys(dimensions)
+            dimensions[String(d)] = dimensions[String(d)][get_slice(slice)]
+        end
     end
 
+    # Writes the sliced dimensions to the specified netcdf file
+    Dataset(filename, mode, attrib=attrib) do ds
+        for (dimname, dimarray) in dimensions
+            defVar(ds, dimname, dimarray, (dimname,),
+                   compression=compression, attrib=dim_attrib[dimname])
+        end
+    end
     return
 end
 
-function write_output_netcdf(fw::NetCDFOutputWriter, fields, iteration)
-    xC, yC, zC = fields["xC"], fields["yC"], fields["zC"]
-    xF, yF, zF = fields["xF"], fields["yF"], fields["zF"]
 
-    u, v, w = fields["u"], fields["v"], fields["w"]
-    T, S    = fields["T"], fields["S"]
+"""
+    NetCDFOutputWriter <: AbstractOutputWriter
 
-    xC_attr = Dict("longname" => "Locations of the cell centers in the x-direction.", "units" => "m")
-    yC_attr = Dict("longname" => "Locations of the cell centers in the y-direction.", "units" => "m")
-    zC_attr = Dict("longname" => "Locations of the cell centers in the z-direction.", "units" => "m")
+An output writer for writing to NetCDF files.
+"""
+mutable struct NetCDFOutputWriter <: AbstractOutputWriter
+             filename :: String
+              dataset :: Any
+              outputs :: Dict
+             interval :: Union{Nothing, AbstractFloat}
+            frequency :: Union{Nothing, Int}
+              clobber :: Bool
+               slices :: Dict
+   len_time_dimension :: Int
```

Comment:
What's this variable for? Is this the best name?

Reply:
It's there because we append data to the netcdf file at each output iteration. So this variable, `len_time_dimension`, is incremented by 1 for every iteration when something is written to the file. It is an internal variable and shouldn't affect the end users.